odoo/fields.py

```diff
@@ -5092,13 +5296,15 @@ def __init__(self, record, field):
 
     def __iter__(self):
         records = self.record.browse(self.record._prefetch_ids)
-        ids = self.record.env.cache.get_values(records, self.field)
-        return unique(id_ for id_ in ids if id_ is not None)
+        context_key = records.env.cache_key(self.field)
+        ids = self.record.env.cache.get_values(self.field, context_key, records._ids)
+        return unique(id_ for id_ in ids if id_ is not None and id_ is not NOTHING)
```

Comment:
Could be simpler, like
```suggestion
        yield from records[field.name]._ids
```